[BUG] Some existing tests fail on Windows

**How: List of performed steps**

```
C:\Users\sekik\esmf-sdk-py-aspect-model-loader\core\esmf-aspect-meta-model-python>poetry run pytest tests\integration\test_cli_functions.py
```

```
C:\Users\sekik\esmf-sdk-py-aspect-model-loader\core\esmf-aspect-meta-model-python>poetry run pytest tests\unit\samm_cli\test_base.py
```

**What: Actual result vs expected result**

Actual results:

```
collected 20 items

 tests/integration/test_cli_functions.py ✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓                                                                                                                            95% █████████▌

―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― TestSammCliErrorHandling.test_nonexistent_file ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

self = <tests.integration.test_cli_functions.TestSammCliErrorHandling object at 0x0000018E93C79BA0>, capfd = <_pytest.capture.CaptureFixture object at 0x0000018E93CBC7F0>
samm_cli = <esmf_aspect_meta_model_python.samm_cli.base.SammCli object at 0x0000018E93C79060>

    def test_nonexistent_file(self, capfd, samm_cli):
        """Test handling of non-existent file."""

        samm_cli.validate("/path/to/nonexistent/file.ttl")

>       assert capfd.readouterr()[1] == "File not found: /path/to/nonexistent/file.ttl\n"
E       AssertionError: assert 'File not fou.../file.ttl\r\n' == 'File not fou...nt/file.ttl\n'
E         Skipping 35 identical leading characters in diff, use -v to show
E         - t/file.ttl
E         + t/file.ttl
E         ?           +

tests\integration\test_cli_functions.py:300: AssertionError

 tests/integration/test_cli_functions.py ⨯                                                                                                                                             100% ██████████
====================================================================================== short test summary info ======================================================================================
FAILED tests/integration/test_cli_functions.py::TestSammCliErrorHandling::test_nonexistent_file - AssertionError: assert 'File not fou.../file.ttl\r\n' == 'File not fou...nt/file.ttl\n'

Results (7.55s):
      19 passed
       1 failed
         - tests\integration/test_cli_functions.py:295 TestSammCliErrorHandling.test_nonexistent_file
```

```
 tests/unit/samm_cli/test_base.py ✓                                                                                                                                                      2% ▎

――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_get_client_path ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

_ = <MagicMock name='_validate_client' id='2625839428464'>, path_mock = <MagicMock name='Path' id='2625839460560'>, join_mock = <MagicMock name='join' id='2625839484880'>

    @mock.patch(f"{BASE_PATH}.join")
    @mock.patch(f"{BASE_PATH}.Path")
    @mock.patch(f"{CLASS_PATH}._validate_client")
    def test_get_client_path(_, path_mock, join_mock):
        base_path_mock = mock.MagicMock(name="base_path")
        base_path_mock.parents = ["parent", "parent_1", "parent_2"]
        path_mock.return_value = path_mock
        path_mock.resolve.return_value = base_path_mock
        join_mock.return_value = "cli_path"

        samm_cli = SammCli()
        result = samm_cli._samm

        assert result == "cli_path"
        path_mock.resolve.assert_called_once()
>       join_mock.assert_has_calls([mock.call("parent", "samm-cli"), mock.call("cli_path", "samm")])

tests\unit\samm_cli\test_base.py:67:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <MagicMock name='join' id='2625839484880'>, calls = [call('parent', 'samm-cli'), call('cli_path', 'samm')], any_order = False

    def assert_has_calls(self, calls, any_order=False):
        """assert the mock has been called with the specified calls.
        The `mock_calls` list is checked for the calls.

        If `any_order` is False (the default) then the calls must be
        sequential. There can be extra calls before or after the
        specified calls.

        If `any_order` is True then the calls can be in any order, but
        they must all appear in `mock_calls`."""
        expected = [self._call_matcher(c) for c in calls]
        cause = next((e for e in expected if isinstance(e, Exception)), None)
        all_calls = _CallList(self._call_matcher(c) for c in self.mock_calls)
        if not any_order:
            if expected not in all_calls:
                if cause is None:
                    problem = 'Calls not found.'
                else:
                    problem = ('Error processing expected calls.\n'
                               'Errors: {}').format(
                                   [e if isinstance(e, Exception) else None
                                    for e in expected])
>               raise AssertionError(
                    f'{problem}\n'
                    f'Expected: {_CallList(calls)}'
                    f'{self._calls_repr(prefix="Actual").rstrip(".")}'
                ) from cause
E               AssertionError: Calls not found.
E               Expected: [call('parent', 'samm-cli'), call('cli_path', 'samm')]
E               Actual: [call('parent', 'samm-cli'), call('cli_path', 'samm.exe')]

..\..\..\AppData\Local\Programs\Python\Python310\lib\unittest\mock.py:966: AssertionError

 tests/unit/samm_cli/test_base.py ⨯✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓✓                                                                                                     100% ██████████
====================================================================================== short test summary info ======================================================================================
FAILED tests/unit/samm_cli/test_base.py::test_get_client_path - AssertionError: Calls not found.

Results (0.81s):
      48 passed
       1 failed
         - tests\unit\samm_cli/test_base.py:52 test_get_client_path
```

Expected results: they should be successful on Windows as well

**Where: (e.g. component, version, url, your system info)**

Component: unit and integration test for SAMM-CLI
Version: main branch
System info: Windows 11 Pro 25H2

--- a/.github/workflows/push_request_check.yml
+++ b/.github/workflows/push_request_check.yml
@@ -21,8 +21,13 @@ concurrency: # Only run one workflow per branch at a time, and cancel any in-pro
 
 jobs:
   build:
-    name: Check if passes all requirements
-    runs-on: ubuntu-latest
+    name: Check if passes all requirements on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: true
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-15-intel]
 
     defaults:
       run:

--- a/core/esmf-aspect-meta-model-python/tests/integration/test_cli_functions.py
+++ b/core/esmf-aspect-meta-model-python/tests/integration/test_cli_functions.py
@@ -297,4 +297,4 @@ class TestSammCliErrorHandling:
 
         samm_cli.validate("/path/to/nonexistent/file.ttl")
 
-        assert capfd.readouterr()[1] == "File not found: /path/to/nonexistent/file.ttl\n"
+        assert capfd.readouterr()[1].rstrip() == "File not found: /path/to/nonexistent/file.ttl"

--- a/core/esmf-aspect-meta-model-python/tests/unit/samm_cli/test_base.py
+++ b/core/esmf-aspect-meta-model-python/tests/unit/samm_cli/test_base.py
@@ -9,6 +9,7 @@
 #
 #   SPDX-License-Identifier: MPL-2.0
 """SammCli unit tests"""
+import platform
 import subprocess
 
 from unittest import mock
@@ -64,7 +65,10 @@ def test_get_client_path(_, path_mock, join_mock):
 
     assert result == "cli_path"
     path_mock.resolve.assert_called_once()
-    join_mock.assert_has_calls([mock.call("parent", "samm-cli"), mock.call("cli_path", "samm")])
+    cli_file = "samm"
+    if platform.system() == "Windows":
+        cli_file += ".exe"
+    join_mock.assert_has_calls([mock.call("parent", "samm-cli"), mock.call("cli_path", cli_file)])
 
 
 @mock.patch(f"{BASE_PATH}.download_samm_cli")

